### ISSUE TITLE
Suggestion: Trainees and Class Randomization 

### ISSUE THREAD

**Issue by SdeSpencer** [lushen124/Universal-FE-Randomizer#91]
Preface: I have no idea how hard this would be and PLEASE tell me if i'm overexplaining something / accidentally being condescending or an idiot or something

It occurred to me that Trainees in Sacred Stones sorta all get the short end of the stick, so i decided to create an idea for a list of options for them that could be added (with the trainee seal being added if necessary)

1. Exclude Trainees
2. Randomize as Tier 0 (only randomize between trainees)
3. Randomize as Tier 1 (treat trainees as if they were Tier 1 for the purposes of randomization, current option)
4. Combine Tier 0 and 1 (Would add base Trainee classes to the randomization list of non-promoted units, otherwise working as 3)

For 1-3, clearly the Promoted trainees would still be in the Tier 1 and Tier 2 randomization lists. For option 4, I'd remove the Tier 1 trainees from the list, to avoid confusion with Tier 0 ones added to the list (as they would be hard to distinguish), but that's up to you.

**Comment by lushen124** [comment]
So just to make sure I understand correctly:

1. Don't touch Ross, Amelia, or Ewan.
2. Ross, Amelia, and Ewan can only randomize into Journeyman, Recruit, and Pupil (Tier 0)
3. Ross, Amelia, and Ewan are considered unpromoted (Tier 1) (which is indeed, the current logic)
4. Journeyman, Recruit, and Pupil (T0 versions) are added into the Tier 1 pool, replacing the usual Tier 1 variants.

I'm a little confused about that last one because that sounds worse off than trainees already are.

**Comment by SdeSpencer** [comment]
That is correct. The last one is more intended as an option so that if people want they can randomize in T0 Trainees as a base option. I'm aware its not great for the current trainees, but i thought it would fit well as a potential option

**Comment by lushen124** [comment]
Part of the goal is to limit the number of characters that need Trainee Seals, since those are a workaround to begin with, as nobody really knows how Trainees are promoted normally. If we can figure out how trainees actually promote and can apply that to non-trainee characters, then this option might make more sense. Options 1 - 3 are fine because they either do less than we do today or at the very least, they don't need Trainee Seals (maybe... option 2 might cause issues since I'm not sure if the game is checking specifically for Journeyman Ross, rather than trainee Ross). I added Trainee Seals because they unlock Ross/Amelia/Ewan for random recruitment (since even Ross as a Journeyman in the wrong character slot is enough to make his autopromotion not work).

**Comment by SdeSpencer** [comment]
Vague connected suggestion, that I'm putting here to avoid Clogging: 

Trainees and Randomized Recruitment.

Only 3 real options would be open (note that option 2 is weird and it'd be ok if you just went "yeah no"):
1. Exclude Trainees (Don't touch Ross, Amelia, or Ewan) 
2. Treat Trainees as T0 (see below)
3. Treat Trainees as T1 (Current option, treats the 3 as if they were Tier 1 without changing their classes at all)

Option 2 Treats Ross, Amelia, and Ewan as T0, bringing them up to to higher tiers if necessary (For Auto-Leveling if they moved up 10 levels if going to a T1 class for instance, or 20 if going to a T2), and characters that can who are in their slots will move down to T0 (same way in reverse for Auto Leveling).
IF Option 2 is selected, and fill class is selected, and the person going into the trainee slot 100% does not have a version of their class in T0, then either (up to you):
1. Do what you do with Special classes without down/up promotions in general (I don't know, i assume they are just excluded from rolling that way)
2. Go to their T1 variant and treat them as if they were randomized into that class with option 3 instead.

**Comment by lushen124** [comment]
The Trainee Seals exist because moving Ross, Amelia, and Ewan off of their spots, but retaining their T0 trainee class will not trigger their auto promotion upon reaching level 10 as T0. It's really a stopgap measure more than anything else.

The current logic considers Trainee slots as T1, but the trainee characters as T0. So, for example, Vanessa randomizing into Ross's slot will remain a T1 pegasus knight (assuming no class randomization) because no trainee class exists for pegasus knight. But Ross randomizing into Vanessa's slot will remain a T0 Journeyman (with an aforementioned Trainee Seal in his inventory). The only time Ross wouldn't remain his T0 Journeyman is if he were to swap places with a promoted unit (Seth, Innes, Dozla, Saleh, Duessel, or Syrene), in which case, he'd become a T2 (Super Journeyman, Hero, Warrior, or Berserker). 

I feel this is closest to FE8 normally is, since Ewan is one of the last unpromoted units you get and he comes relatively late in the game, I figure all unpromoted units could be a trainee.

That said, Option 1 can be implemented by just lumping Trainee classes as "special classes" and piggyback off the same option, so that one is easy. The current norm is a cross between your Option 2 and Option 3.

**Comment by SdeSpencer** [comment]
Wait wait wait I think you misunderstand. my point of what my Option 3 was WAS that "mix of Option 2 and 3" in that they can switch with T1 classes at will without changing their class (effectively, they would ignore that their current classes and slots were T0), while Option 2 would treat them fully as T0 and move them up (and other people down) as such. Also personally it might be weird to put them as "special classes" as they have sorta weird choices, but it is up to you. And again, personally i'd say that diversifying your use of that Stopgap measure of Trainee seals could be a good idea.

**Comment by lushen124** [comment]
Are you trying to separate those two into distinct options? Where one option treats them entirely as T1 and the other... well I guess the other does the same as it does today, where they remain T0s.

>  while Option 2 would treat them fully as T0 and move them up (**and other people down**) as such.

If I'm understanding this correctly, are you suggesting normally T1 characters move down into T0?

**Comment by SdeSpencer** [comment]
No??? 

Option 2 would treat Ross, Amelia, and Ewan as T0, So 
1. when moving to T1 they'd increase to a T1 class they could normally promote to (assuming Fill class), and would be treated as moving up 10 levels (assuming Autolevel Base Stats)
2.When moving to T2 they'd increase to a T2 class they could normally promote to (assuming Fill class), and would be treated as moving up 20 levels (assuming Autolevel Base Stats)
3. When others moved to their slot they'd move down to T0, doing all that in reverse (if possible, which is why i gave options for what to do in such a situation earlier)

Option 3 would treat Ross, Amelia, and Ewan as T1, So
1. When moving to T1, they'd remain their current class (assuming Fill class) and not change base stats (assuming Autolevel Base Stats)
2. When moving to T2, they'd increase to a T2 class they could normally promote to (assuming Fill class), and would be treated as moving up 10 levels (assuming Autolevel Base Stats)
3. When others moved to their slot they'd act as if moving into a T1 slot, with all the aspects as such.

Option 3 is what you currently do isn't it??

**Comment by lushen124** [comment]
Option 3 is exactly what I do. 

If I'm reading Option 2 correctly:

* If Ross randomizes into Amelia's slot, then Ross is a T0 Journeyman.
* If Ross randomizes into Vanessa's slot, then Ross is a T1 Journeyman/Fighter/Pirate.
* If Ross randomizes into Duessel's slot, then Ross is a T2 Journeyman/Warrior/Hero/Berserker.
* If Vanessa randomizes into Ross/Amelia/Ewan's slot, then Vanessa is a T1 Pegasus Knight.

And to me, this is the interesting piece:

* If Garcia randomizes into Ross/Amelia/Ewan's slot, then Garcia is a T0 Journeyman.
* If Dozla randomizes into Ross/Amelia/Ewan's slot, then Dozla is a T0 Journeyman.

**Comment by SdeSpencer** [comment]
Yes! That is exactly what i was thinking.

Small Note: It's up to you if you want Vanessa to be ABLE to randomize into Ross/Amelia/Ewan's Slot, but from what i gather that is what you prefer.
